### PR TITLE
ViewLocator param name should follow the interface

### LINKED ISSUE #269: ViewLocator causes warning CA1725
### Describe the bug

The template (as well as all code examples I can find) for the default ViewLocator implements `ITemplate.Build` with this signature:
```cs
Control? Build(object? data)
```
but the interface has a different parameter name:
```cs
Control? Build(object? param)
```

This causes warning [CA1725: Parameter names should match base declaration](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1725)

Depending on your code analysis settings, this can cause the build to fail.

### To Reproduce

1. Create a project from the `avalonia.mvvm` template.
2. Add these properties to the project:
  ```xml
  <EnableNETAnalyzers>True</EnableNETAnalyzers>
  <AnalysisLevel>8.0-recommended</AnalysisLevel>
  <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
  ```
3. build the project

The build fails with:
```
ViewLocator.cs(11,39): error CA1725: In member Control? ViewLocator.Build(object? data), change parameter name data to param in order to match the identifier as it has been declared in Control? ITemplate<object?, Control?>.
Build(object? param) (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1725)
```

### Avalonia Templates version

11.0.10.1

### Avalonia version

11.1.3

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by maxkatz6**
Closes #269

### CODE REVIEW

File: templates/csharp/app-mvvm/ViewLocator.cs [hunk 1]
```diff
@@ -8,19 +8,17 @@ namespace AvaloniaAppTemplate;
...
+        var name = param.GetType().FullName!.Replace("ViewModel", "View", StringComparison.Ordinal);
         var type = Type.GetType(name);
 
         if (type != null)
         {
-            var control = (Control)Activator.CreateInstance(type)!;
-            control.DataContext = data;
-            return control;
+            return (Control)Activator.CreateInstance(type)!;
         }
```
Comment (MalzSmith) [inline templates/csharp/app-mvvm/ViewLocator.cs:22]: Hey, is there a reason for making this change? The F# template still has the previous default behavior (which I personally prefer)

Comment (maxkatz6) [reply templates/csharp/app-mvvm/ViewLocator.cs:22]: This change was made to match our other template: https://github.com/AvaloniaUI/avalonia-dotnet-templates/blob/main/templates/csharp/xplat/AvaloniaTest/ViewLocator.cs

Also, hardcoding DataContext in this code potentially can be problematic. Typically, DataContext should be inherited from parent instead.

Comment (maxkatz6) [reply templates/csharp/app-mvvm/ViewLocator.cs:22]: I didn't change F# template, as I can't confidently code in F# or test it. 
If you think it should also be updated in any way, and can test it, please open a PR, I can review it at least.

Comment (MalzSmith) [reply templates/csharp/app-mvvm/ViewLocator.cs:22]: Yeah, this makes sense, thanks for getting back to me. I was so happy about being a contributor with #238 but I understand your reasoning.

**Pull request merged**